examples/gno.land/r/stackdump/home/index.gno

```diff
@@ -0,0 +1,86 @@
+package home
+
+import (
+	"std"
+	"strings"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ufmt"
+)
+
+var (
+	realmAllowPrefix = []string{}
+	registry         = avl.NewTree()
+	render           []string
+)
+
+func SetOutput(s ...string) {
+	AssertAdminAccess()
+	render = []string{}
+	appendOutput(s...)
+}
+
+func appendOutput(s ...string) {
+	for _, v := range s {
+		render = append(render, v)
+	}
+}
+
+func init() {
+	realmAllowPrefix = append(realmAllowPrefix, std.CurrentRealm().PkgPath()+"/patch")
+	appendOutput("foo", "bar")
+	registry.Set("foo", "testing!")
+	registry.Set("bar", "\n123")
+}
+
+/* Auth API */
+func HasAllowedPrefix() bool {
+	//if std.IsUser() {
+	//    return false
+	//}
+	prevPath := std.PreviousRealm().PkgPath()
+	if prevPath != "" {
+		return false
+	}
+	currentPath := std.CurrentRealm().PkgPath()
+	for _, callerPath := range realmAllowPrefix {
+		if strings.HasPrefix(currentPath, callerPath) {
+			return true
+		}
+	}
+	return false
+}
+
+func AssertAdminAccess() {
+	if !HasAllowedPrefix() {
+		panic("access denied: " + std.PreviousRealm().PkgPath() +
+			" realm must match an allowed prefix:[" + strings.Join(realmAllowPrefix, ",") + "]" +
+			" got: " + std.CurrentRealm().PkgPath())
+	}
+}
+
+func Set(key string, val interface{}) {
+	// AssertAdminAccess()
+	registry.Set(key, val)
+}
```

Comment:
NOTE: no crossing call here ..

When we did add it - the correct behavior is observed - even when called from inside Init()
```
missing cross before external crossing()
```